.github/workflows/check-links-cron.yaml

```
@@ -0,0 +1,55 @@
+name: Periodically Check Documentation Links
+
+on:
+  schedule:
+    - cron: '0 9 * * 1'
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - '.github/workflows/check-links-cron.yaml'
+
+permissions: read-all
+
+jobs:
+  links-checker:
+    name: Documentation links checker
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          persist-credentials: false 
+      - name: Check Documentation links
+        run: |
+          ./Documentation/check-links.sh

```

I agree that the time elapsed for the test is not realistic. The script didn't run correctly, we could try a few things to check it:

- Run with `bash -x` to check where it exits
- Remove the `-q` option passed to Sphinx in `check-links.sh`. Actually, we should maybe do this even after we figured the issue. This would show that we effectively test the links, show how many rate-limit occur, or any warning that we skip, as well that all the redirects that happen.

My guess is simply that Sphinx (and the linkchecker) are not installed on the runner. We should run this command from the `docs-builder` container, which has all dependencies set up, just like we do for the documentation builds in CI (`documentation.yaml`):
```suggestion
      - name: Check Documentation links
        uses: docker://quay.io/cilium/docs-builder:4b194c0facc5d1d3b9c0dfef3529ee7c661a6b73@sha256:095cf9621cf46c6c878e25c3a98f02ef3d0aed1154936663e352fd3fe5e7613d
        with:
          entrypoint: ./Documentation/check-build.sh
```

---

I think this will require us to make changes to the script as well ?

---

My bad, obviously I meant `check-links.sh` and not `check-build.sh`. I realised my mistake after having logged off yesterday but forgot to update in the evening :disappointed:.